packages/react-components/src/shared/src/component-types.ts

```diff
@@ -0,0 +1,28 @@
+import { ElementType, ForwardedRef, PropsWithChildren, ReactNode } from "react";
+
+export interface ForwardedRefAttribute<T = any> {
```

Comment:
Generics types won't be used, or they will only be used with the default value. However, adding them to the types describe the type better than 

```
export interface ForwardedRefAttribute {
    /**
     * @ignore
     */
    forwardedRef: ForwardedRef<any>
}
```